In [2]:
import pandas as pd
import numpy as np

In [3]:
#Read dataset
df_avocado = pd.read_csv('avocado.csv')

In [4]:
#read information of dataset
df_avocado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18249 entries, 0 to 18248
Data columns (total 14 columns):
Unnamed: 0      18249 non-null int64
Date            18249 non-null object
AveragePrice    18249 non-null float64
Total Volume    18249 non-null float64
4046            18249 non-null float64
4225            18249 non-null float64
4770            18249 non-null float64
Total Bags      18249 non-null float64
Small Bags      18249 non-null float64
Large Bags      18249 non-null float64
XLarge Bags     18249 non-null float64
type            18249 non-null object
year            18249 non-null int64
region          18249 non-null object
dtypes: float64(9), int64(2), object(3)
memory usage: 1.9+ MB


In [15]:
df_avocado.describe().T

count           mean           std      min       25%  \
Unnamed: 0    18249.0      24.232232  1.548104e+01     0.00     10.00   
AveragePrice  18249.0       1.405978  4.026766e-01     0.44      1.10   
Total Volume  18249.0  850644.013009  3.453545e+06    84.56  10838.58   
4046          18249.0  293008.424531  1.264989e+06     0.00    854.07   
4225          18249.0  295154.568356  1.204120e+06     0.00   3008.78   
4770          18249.0   22839.735993  1.074641e+05     0.00      0.00   
Total Bags    18249.0  239639.202060  9.862424e+05     0.00   5088.64   
Small Bags    18249.0  182194.686696  7.461785e+05     0.00   2849.42   
Large Bags    18249.0   54338.088145  2.439660e+05     0.00    127.47   
XLarge Bags   18249.0    3106.426507  1.769289e+04     0.00      0.00   
year          18249.0    2016.147899  9.399385e-01  2015.00   2015.00   

                    50%        75%          max  
Unnamed: 0        24.00      38.00        52.00  
AveragePrice       1.37       1.66         3.25  
Total Volume  107376.76  432962.29  62505646.52  
4046            8645.30  111020.20  22743616.17  
4225           29061.02  150206.86  20470572.61  
4770             184.99    6243.42   2546439.11  
Total Bags     39743.83  110783.37  19373134.37  
Small Bags     26362.82   83337.67  13384586.80  
Large Bags      2647.71   22029.25   5719096.61  
XLarge Bags        0.00     132.50    551693.65  
year            2016.00    2017.00      2018.00

Average price is $1.4, minmum price and maxmum price being $0.44 and $3.25.

In [5]:
#check the feature of dataset
df_avocado.head()

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0           0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1           1  2015-12-20          1.35      54876.98   674.28   44638.81   
2           2  2015-12-13          0.93     118220.22   794.70  109149.67   
3           3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4           4  2015-11-29          1.28      51039.60   941.48   43838.39   

     4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
0   48.16     8696.87     8603.62       93.25          0.0  conventional   
1   58.33     9505.56     9408.07       97.49          0.0  conventional   
2  130.50     8145.35     8042.21      103.14          0.0  conventional   
3   72.58     5811.16     5677.40      133.76          0.0  conventional   
4   75.78     6183.95     5986.26      197.69          0.0  conventional   

   year  region  
0  2015  Albany  
1  2015  Albany  
2  2015  Albany  
3  2015  Albany  
4  2015  Albany

In [16]:
#Check if null value exsist
df_avocado.isnull().any()

Unnamed: 0      False
Date            False
AveragePrice    False
Total Volume    False
4046            False
4225            False
4770            False
Total Bags      False
Small Bags      False
Large Bags      False
XLarge Bags     False
type            False
year            False
region          False
dtype: bool

There is no null data, we don't have to deal with nan values.

In [18]:
#sort the data by date
df = df_avocado.sort_values('Date')

#Select the columns that categor
df.head()

Unnamed: 0        Date  AveragePrice  Total Volume     4046      4225  \
11569          51  2015-01-04          1.75      27365.89  9307.34   3844.81   
9593           51  2015-01-04          1.49      17723.17  1189.35  15628.27   
10009          51  2015-01-04          1.68       2896.72   161.68    206.96   
1819           51  2015-01-04          1.52      54956.80  3013.04  35456.88   
9333           51  2015-01-04          1.64       1505.12     1.27   1129.50   

          4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
11569   615.28    13598.46    13061.10      537.36          0.0       organic   
9593      0.00      905.55      905.55        0.00          0.0       organic   
10009     0.00     2528.08     2528.08        0.00          0.0       organic   
1819   1561.70    14925.18    11264.80     3660.38          0.0  conventional   
9333      0.00      374.35      186.67      187.68          0.0       organic   

       year              region  
11569  2015           Southeast  
9593   2015             Chicago  
10009  2015  HarrisburgScranton  
1819   2015          Pittsburgh  
9333   2015               Boise

In [19]:
#Save the new dataset file to 'df_avocado.csv'
df.to_csv('df_avocado.csv')

This dataset's data is quite clear, no null or missing data. Saved the new dataset file to 'df_avocado.csv' to finisbed the data wrangling work.